src/position.cpp

```diff
@@ -923,8 +923,13 @@ Bitboard Position::attackers_to(Square s, Bitboard occupied) const {
 Bitboard Position::attackers_to_pseudo_royals(Color c) const {
   Bitboard attackers = 0;
   Bitboard pseudoRoyals = st->pseudoRoyals & pieces(~c);
+  Bitboard pseudoRoyalsTheirs = st->pseudoRoyals & pieces(c);
   while (pseudoRoyals) {
       Square sr = pop_lsb(pseudoRoyals);
+      if (blast_on_capture()
+          && (pseudoRoyalsTheirs & attacks_bb<KING>(sr)) == pseudoRoyalsTheirs)
```

Comment:
By the definition of st->pseudoRoyals it is already considered that losing **one** of them would lose the game, see https://github.com/ianfab/Fairy-Stockfish/blob/67af9946b06c73eace9bbe7dcf771204cd7cb304/src/position.cpp#L543-L553
Therefore, it should be sufficient here to check that `(pseudoRoyalsTheirs & attacks_bb<KING>(sr))` is truthy.

Reply:
I see. So pseudo-royalness only applies for the last piece before extinction kicks in anyway.